### PR TITLE
Add tag moderators on users page (for admins)

### LINKED ISSUE #20405: Improve assigning tag moderators (for admins)
There are a couple of issues related to the users search in /admin.
E.g. when adding a tag moderator on `/admin/content_manager/tags/{id}/edit` page.

1) We are currently searching by the specified string in username, name and email with `ILIKE %string%` (see `Admin::UsersQuery.search_relation`) which often can lead to a large number of search results (e.g. when we search for a short  or popular name/word). We also limit the results to 10, so the actual user we search for may not appear in the results at all.

There is an [additional filtering](https://github.com/forem/forem/blob/473594f1920966cc40d749141d7ca44b740eea78/app/javascript/shared/components/UserStore.js#L27C26-L41) on the frontend which is slightly different (e.g. it's case sensitive, and the search on the backend is case-sensitive), that's why we get less that 10 results.

**Possible solution:**
- make search "stricter", use case-sensitive search and/or use strict matching, not just inclusion of the search string
- search only by username (the label says "enter the username")
- remove filtering on the frontend or change it so that it wouldn't contradict the backend one

**Concern:**

The same code is used in other places: users list (search) in member manager and `/admin/invitations`, so if we make changes we need to make sure that search in those places keeps working as intended.

2) It's not clear what user we select from the search, because we show users `name`, not `username` in the search results. It may be misleading because names are not unique. It may lead to assigning wrong users as tag moderators.

**Solution:** show both names and usernames (or only unuque username) in the search results dropdown.

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
Implemented adding tag moderators from member manager page for admins.
This feature was "stubbed" for a long time and the tag mods could be only added from the tag page which is not always convenient.

Add tags one at a time.

## Related Tickets & Documents
- Closes #20405

## QA Instructions, Screenshots, Recordings
Visit member manager page for admins, go to users page.
You can now add tag moderator roles in the "tags" block and remove them.

изображение

## Added/updated tests?

- [x] Yes

### CODE REVIEW

File: app/controllers/admin/tags/moderators_controller.rb [hunk 1]
```diff
@@ -18,16 +18,14 @@ def create
           return redirect_to edit_admin_tag_path(params[:tag_id])
         end
 
-        notification_setting = user.notification_setting
-        if notification_setting.update(email_tag_mod_newsletter: true)
-          TagModerators::Add.call([user.id], [params[:tag_id]])
-          flash[:success] =
-            I18n.t("admin.tags.moderators_controller.added", username: user.username)
+        result = TagModerators::Add.call(user.id, params[:tag_id])
```
Comment (lightalloy) [inline app/controllers/admin/tags/moderators_controller.rb:21]: Moved updating notifications settings to a service object to reuse the service object in `Admin::UsersController`.

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -107,14 +108,13 @@ def destroy
...
-      resource_type = params[:resource_type]
-
       @user = User.find(params[:user_id])
 
       response = ::Users::RemoveRole.call(
         user: @user,
         role: role.name,
-        resource_type: resource_type,
+        resource_type: params[:resource_type],
+        resource_id: params[:resource_id],
```
Comment (lightalloy) [inline app/controllers/admin/users_controller.rb:117]: Added `resource_id` to be able to delete role for a particular tag.

File: app/services/tag_moderators/add.rb [hunk 3]
```diff
@@ -1,18 +1,21 @@
...
+    Result = Struct.new(:success?, :errors, keyword_init: true)
+
+    def self.call(user_id, tag_id)
+      new(user_id, tag_id).call
     end
 
-    def initialize(user_ids, tag_ids)
-      @user_ids = user_ids
-      @tag_ids = tag_ids
+    def initialize(user_id, tag_id)
```
Comment (lightalloy) [inline app/services/tag_moderators/add.rb:9]: Simplified the class: we are not using adding multiple tag mods at once, also the code was a bit counterintuitive.

**Review by benhalpern — APPROVED** [review]
lgtm

**Pull request merged**